### DISCUSSION TITLE
Use assets in loading of other assets

### DISCUSSION THREAD

**Discussion by yngwi** [bevyengine/bevy#5674] [Q&A] [answered]
Hi,

I wonder how I can do the following thing: 

1. I would like to load a configuration yaml (using [bevy_common_assets](https://github.com/NiklasEi/bevy_common_assets/)) from the assets folder that includes, among other configuration data the name of the texture atlas to be used for the 2d display of my app. 
2. I would then like to use this name once it is loaded to load the specified texture atlas properties file that contains data about rows, columns, sprite size as well as the name of the actual sprite file. 
3. Then I would use this data to actually create the texture atlas with the `TextureAtlas::from_grid_with_padding` method and `asset_server.load(&atlas_name)`. 

I couldn't find an example on how to do this so I am grateful for any pointers. Thank you!

Edit: I also generally wonder how you can use data from assets during the initialization phase. I have another use case, for instance: I would like to load a list of colors from another yml file and use this colors then to, for instance, set the clear color during app setup. Is there a way to crate a startup system that waits until specific assets are loaded?

Daniel

**Comment by kgv** [comment, marked as answer]
One possible solution is to wait until the required assets are loaded. After that, initialize the dependent data.
Checking the loading process is done using `LoadState` ([example](https://bevy-cheatbook.github.io/assets/ready.html)).

**Reply by yngwi** [reply]
thank you!
